ouroboros-network/protocol-tests/Ouroboros/Network/Protocol/Handshake/Test.hs

```diff
@@ -447,6 +452,99 @@ prop_pipe_IO :: ArbitraryVersions -> Property
 prop_pipe_IO (ArbitraryVersions clientVersions serverVersions) =
     ioProperty (prop_channel createPipeConnectedChannels clientVersions serverVersions)
 
+--
+-- Asymmetric tests
+--
+
+
+-- | Run a simple handshake client and server using connected channels.
+-- The server can only decode a subset of versions send by client.
+-- This test is using a fixed serverver 'Versions' which can only accept
+-- a single version 'Version_2' (it cannot decode any other version).
```

Comment:
Looking at `versionNumberCodec'` it looks like the single version should be `Version_1`.

Reply:
You're right.